### PR TITLE
docs(guides): update ibis-for-dplyr-users.ipynb with latest

### LINKED ISSUE #6125: docs: Fix my bad rebase of Ibis for dplyr users doc
I goofed up when squashing in https://github.com/ibis-project/ibis/pull/6050 which resulted in an out of date copy of the doc getting merged and made visible at https://ibis-project.org/ibis-for-dplyr-users/. I'm hoping I can comb through my reflog tomorrow to recover everything and will update here with how that goes.

### PR DESCRIPTION
**Pull request by amoeba**
Fixes #6125

### CODE REVIEW

**Comment by amoeba** [comment]
Hey @cpcloud or @gforsyth can you take a look here? As I mentioned in https://github.com/ibis-project/ibis/issues/6125, I totally messed up my squash and didn't catch my mistake until https://github.com/ibis-project/ibis/pull/6050 was merged. I've successfully pulled the correct copy out of the reflog. While reviewing locally, I found a few issues which are also included here:

- The TOC was actually broken and I hadn't noticed so not all headers were showing. I found the cause and fixed it so all headers show in the TOC
- The breaking change introduced in 3caf3a12469d017428d5e2bb94143185e8770038 that changes how suffixes work in joins affected this notebook and I've updated the text and code to match the new behavior
- I updated two headers to match the same style as the previous ones so the headers are all consistent now
- One cell had slightly quirky formatting which I've fixed

I don't think any of these changes will be controversial. I've checked that the output is right locally and on ReviewNB. Sorry for the mistake in the first place.

File: docs/ibis-for-dplyr-users.ipynb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
```
Comment (cpcloud) [inline docs/ibis-for-dplyr-users.ipynb:1]: <p>Did you mean to give this a large bold style?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis/pull/6129/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (cpcloud) [inline docs/ibis-for-dplyr-users.ipynb:1]: <p>Small typo in <code>introudction</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis/pull/6129/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='71'/>

**Comment by cpcloud** [comment]
@amoeba This looks really great! Thanks for following up.

Couple of micronits, but nothing blocking.

File: docs/ibis-for-dplyr-users.ipynb (hunk 1 above)
Comment (amoeba) [reply docs/ibis-for-dplyr-users.ipynb:1]: <p>Good catch. Fixed.</p>

Comment (amoeba) [reply docs/ibis-for-dplyr-users.ipynb:1]: <p>I don't follow, can you explain what you mean here?</p>

**Comment by amoeba** [comment]
Thanks @cpcloud, left you one followup question.

File: docs/ibis-for-dplyr-users.ipynb (hunk 1 above)
Comment (cpcloud) [reply docs/ibis-for-dplyr-users.ipynb:1]: <p>This shows up weird in the ReviewNB UI, but is fine in the rendered docs, it's good to go!</p>

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
Thanks @amoeba !

**Pull request merged**